walletconnectv2/src/main/kotlin/com/walletconnect/walletconnectv2/client/mapper/ClientMapper.kt

```
@@ -73,6 +72,23 @@ internal fun EngineDO.DeletedSession.toClientDeletedSession(): WalletConnect.Mod
 internal fun EngineDO.SessionNotification.toClientSessionNotification(): WalletConnect.Model.SessionNotification =
     WalletConnect.Model.SessionNotification(topic, type, data)
 
+@JvmSynthetic
+internal fun EngineDO.SettledPairing.toClientSettledPairing(): WalletConnect.Model.SettledPairing =
+    WalletConnect.Model.SettledPairing(
+        topic.value, permissions = WalletConnect.Model.SessionPermissions(
+            WalletConnect.Model.Blockchain(permissions.blockchain?.chains ?: emptyList()),
+            WalletConnect.Model.Jsonrpc(permissions.jsonRpc?.methods ?: emptyList())
```

should we default to empty lists for blockchains and jsonRpc? Or should we throw an exception if these are not present

---

This is good point. One of my next goals/tasks is to validate all the messages and permissions and throw errors when necessary.
I'll add TODO for now.